### ISSUE TITLE
Rename the repo to something like td-ameritrade-java-client-React so it doesn't have a duplicate name.

### ISSUE THREAD

**Issue by studerw** [preintercede/TD-Ameritrade-Java-Client#1]
Also, are you going to do definitely use React? If you want to learn front-end, it seems either React, Vue, or Angular are the best choices as far as employment is concerned. 

I have a [React Udemy course](https://www.udemy.com/course/react-the-complete-guide-incl-redux/) you could use (if I can share it somehow) it is really good! 

If you do React, re-create the repo with the new name, and then use something like [React-Create-App](https://create-react-app.dev/docs/getting-started/) to make sure you can get a dev environment up. I would suggest either Visual Studio Code or Webstorm for an IDE, but obviously that is your choice, too. Let me know if you run into any issues.

**Comment by studerw** [comment]
For the certificate:

Once you download it and install it and try to go to the page, you will see this:

![image](https://user-images.githubusercontent.com/6321283/69271791-852b3e80-0ba3-11ea-908c-aa2bcf0183d0.png)

To install, it looks like you follow [these directions](https://support.globalsign.com/customer/portal/articles/1211541-install-client-digital-certificate---windows-using-chrome).

Do NOT add the certificate to Git repo. Because then everyone else can connect to the site - it is kind of like accidentally leaving your password and putting it on the internet :(.

The way to do that is to either just not add it to your repo folder at all, or you can do so but add the filename to a [.gitignore](https://help.github.com/en/github/using-git/ignoring-files) file.

**Comment by preintercede** [comment]
I'm use React but no guarantees that I might contribute since this looks intimidating, but I'll give it a shot. 

I'm actually taking a react class on Udemy right now. 
https://www.udemy.com/course/react-redux/

I've only finished up to the part right before redux. 

So I should first download the repo that you linked and make a new react app and then start from there right?

**Comment by studerw** [comment]
You don't need to download any repo of mine because it is backend stuff. What you need to do is the following:

1. Download that certificate (keep it private). Make sure you can get to [this page](https://tda-demo.eastus.cloudapp.azure.com:8443/swagger-ui.html) and use the cert in your browser to connect. 
2. In the page, see if you understand the docs and how they are REST calls. The easiest one to try in the browser is 'tda-controller.ping'. You just make a get request, and it sends a 'pong' back to you.
3. In your new repo, do the [Create React App](https://create-react-app.dev/), add a README with a sentence or two describing your goals, and then check it into a new REPO in github (you could use this one, just rename it so it isn't a copy of my backend repo - too confusing).

**Comment by studerw** [comment]
hopefully you can understand my English if it's not your 1st language. My Spanish is okay, and German is horrible, so English is probably our best bet ;)

**Comment by preintercede** [comment]
Don't worry, English is my first language. Just ignore that grammar error I made up top. :p

I think I got it working since I can connect to the site. Is there a goal I should be doing? Am I building something with React. I'm still new to this so it's super confusing to me.

**Comment by studerw** [comment]
ok, your main goal is to learn, not to actually get anything working for me, though if you can help me debug, that's a bonus for me :).

So go at your own pace. 

Are you completely new to all development, or just front-end or what. What's you background? And what's you goal here? To get a job or to learn or what?

**Comment by preintercede** [comment]
Well goal is to learn and help contribute to your project so that maybe I could put it on my resume.

As for me, I graduated with a non-CS degree and watched to switch. I have basic knowledge of Java and Python and I'm currently learning React to expand my skillset. My goal is to find a job, but it's not looking so good since I get ghosted or rejected.

**Comment by studerw** [comment]
OKay cool - let's make a goal to get a really cool project with 'SCREENSHOTS!!!'. That always impresses the HR drones, lol. This one will be easy to do that because we can use a REACT toolkit to make pretty graphs like this: 
![image](https://user-images.githubusercontent.com/6321283/69278100-9f6b1980-0baf-11ea-87c7-7f07ba1a52ce.png).

So again, try to run that 'create-react-app' project and get it working on your local machine. Then I will show you how to make a call to the rest service and we'll go from there.

**Comment by studerw** [comment]
We will use a call like this: 
[Get PriceHistory for Microsoft](https://tda-demo.eastus.cloudapp.azure.com:8443/api/v1/tda/priceHistory?symbol=msft).

That will return a bunch of JSON in your browser. You should download this chrome extension called [JSON Formatter](https://chrome.google.com/webstore/detail/json-formatter/bcjindcccaagfpapjjmafapmmgkkhgoa?hl=en) and then you will get a result like this:

![image](https://user-images.githubusercontent.com/6321283/69278436-55cefe80-0bb0-11ea-9802-b7b86b816d28.png)

**Comment by preintercede** [comment]
I already made my create-react-app. I usually delete all the files under source and make a new file called index.js and get it to show something on the localhost. Should I be doing the same here or should I keep all the files in?

**Comment by studerw** [comment]
You should check in everything that is needed to run the project, but make sure to put anything that is private in .gitignore (like the certificate file). Then in your README, make sure you have a section called 'Build' or something like that with instructions (which will be 'npm start', probably). Make sure to list what version of NPM you're using, too so that everyone else get's duplicate environment.

**Comment by preintercede** [comment]
I'm a bit confused about the certificate part. The website you gave me runs after I downloaded the cert that you pmed me.

**Comment by studerw** [comment]
did you follow the directions to install the cert into Windows using certmgr? 

Once you install it, it probably won't ask you again in Chrome. If you have Firefox and go to the same site, it won't let you in since you have to install the cert manually in Firefox.

**Comment by preintercede** [comment]
Yup I added it. What should my next steps be?

**Comment by preintercede** [comment]
The GetPriceHistory for Microsoft crashes my Chrome after a few seconds.

**Comment by studerw** [comment]
Haha, maybe it is too much data. Try this call: https://tda-demo.eastus.cloudapp.azure.com:8443/api/v1/tda/fundamentals?symbol=msft

does that work okay?

**Comment by preintercede** [comment]
Yup that works fine!

**Comment by studerw** [comment]
okay whenever you are ready check in your project (Not the cert, of course) and I will take a look and make some PRs for you to merge. Let me know if you need help with understanding Git merges and anything else. It is really confusing at first.

**Comment by preintercede** [comment]
I uploaded what I have so far, not sure if I should have done more.

**Comment by studerw** [comment]
Have you written any JavaScript before? React is just adding complexity if not, so we can make it easier for you to start. Check in the most basic html page call hello.html with a single input and and a button, and when you click the button, do an alert that says ‘hello’

**Comment by preintercede** [comment]
I kinda have written JS before, but not an actual program. Basically codewars exercise JS. I'll do my best to make the app. Basically just make like a search bar with a button and when press the button, hello pops up right?

**Comment by studerw** [comment]
Yeah just make the most minimum HTML5 page, an input, a button, and the minimum JS to say hello. We will forget about the React for now until you understand the basics :)

**Comment by preintercede** [comment]
I made the input and  enter button, but still new to the event actions. Should I do onButton pressed in a new js file or same file as inputterm?

**Comment by studerw** [comment]
Keep everything in the same html file for now to keep it easy - you just create a script section somewhere to add the JS and that way everything is in one file, very simple. The event to add to button is ‘onClick’.

**Comment by studerw** [comment]
If you get stuck, then check in to GIT what you have, and then I will look and fix it and send you a merge request. this is how you learn how to deal with GIT.

if you get overwhelmed, don’t worry - it seems like a lot of stuff, but once you learn it well, it will make sense.

**Comment by preintercede** [comment]
I thought I learned a bit from udemy react courses, but building an app is harder  than it looks. 

I'm unsure on how to make the popup. I uploaded what I have to git so far.

**Comment by studerw** [comment]
Okay I’ll take a look a bit later. The easiest way to do an alert is just to call ‘alert(‘hello’).

unfortunately, that function has been abused so much for 20 years with pop ups that sometimes browsers block it automatically so you have to pay attention to what Chrome is doing.

Haha you’ll learn soon enough if you want to spend the rest of your life writing JS. It pays well, but most people don’t like doing it 50 hours a week. Sometimes I wonder if I should selfish careers too.

**Comment by preintercede** [comment]
Oh, I guess I was googling the wrong things since I was looking for popups instead of alerts. 

It's fixed now though!

**Comment by studerw** [comment]
Did you push it (I.e. check into git)?

**Comment by preintercede** [comment]
Do you mean through the terminal? I updated it manually.
